packages/system/src/style.ts

```diff
@@ -224,26 +244,30 @@ export const compose = <TProps = {}>(
   const getStyle = (props: Props<Theme>, sort = true) => {
     const styles = {} as CSSObject
 
+    let generator
     for (const key in props) {
-      const generator = generatorsByProp[key]
+      generator = generatorsByProp[key]
       if (generator) {
         const style = generator.meta.getStyle(props, false)
         merge(styles, style)
       }
     }
 
-    if (!sort) return styles
+    if (!generator || !sort) return styles
 
     const medias = getCachedVariants(props, getCache(props.theme, '__states'))
     return sortStyles(styles, medias)
   }
 
-  const props = flatGenerators.reduce(
-    (allProps, generator) => [...allProps, ...generator.meta.props],
-    [] as string[],
-  )
+  let props = [] as string[]
+  let cssGetters = {} as { [key: string]: ThemeGetter }
+  for (let i = 0; i < flatGenerators.length; i++) {
+    const generator = flatGenerators[i]
+    props = [...props, ...generator.meta.props]
+    cssGetters = { ...cssGetters, ...generator.meta.cssGetters }
+  }
```

Comment:
Since we've moved away from the functional approach here anyway, we could consider direct push and assign for performance, instead of making copies each time through the loop:

```javascript
const props = [] as string[]
const cssGetters = {} as { [key: string]: ThemeGetter }
for (const generator of flatGenerators) {
  props.push(...generator.meta.props)
  Object.assign(cssGetters, generator.meta.cssGetters)
}
```